janitor/functions/_numba.py

```diff
@@ -29,53 +29,55 @@ def _numba_single_join(
         # for the not equal join, we combine indices
         # from strictly less than and strictly greater than indices
         # as well as indices for nulls, if any
-        left_nulls, right_nulls = _numba_not_equal_indices(left_c, right_c)
+        left_nulls, right_nulls = _numba_not_equal_indices(left, right)
         dummy = np.array([], dtype=int)
-        result = _numba_less_than_indices(left_c, right_c)
+        result = _numba_less_than_indices(left, right)
         if result is None:
             lt_left = dummy
             lt_right = dummy
         else:
             lt_left, lt_right = _numba_generate_indices_ne(
                 *result, strict, keep, op_code=1
             )
-        result = _numba_greater_than_indices(left_c, right_c)
+        result = _numba_greater_than_indices(left, right)
         if result is None:
             gt_left = dummy
             gt_right = dummy
         else:
             gt_left, gt_right = _numba_generate_indices_ne(
                 *result, strict, keep, op_code=0
             )
-        left_c = np.concatenate([lt_left, gt_left, left_nulls])
-        right_c = np.concatenate([lt_right, gt_right, right_nulls])
-        if (not left_c.size) & (not right_c.size):
+        left = np.concatenate([lt_left, gt_left, left_nulls])
+        right = np.concatenate([lt_right, gt_right, right_nulls])
+        if (not left.size) & (not right.size):
             return None
         if keep == _KeepTypes.ALL.value:
-            return left_c, right_c
-        indexer = np.argsort(left_c)
-        left_c, pos = np.unique(left_c[indexer], return_index=True)
+            return left, right
+        indexer = np.argsort(left)
+        left, pos = np.unique(left[indexer], return_index=True)
         if keep == _KeepTypes.FIRST.value:
```

Comment:
Like this we could change to `if keep == 'first':`

Reply:
so basically get rid of `enum` and just use the strings directly?

Reply:
Yes, `Enum` is a little bit heavy for `keep`. `keep` only has three values. `Enum` is used to build a relationship between string and number, in my opinion.
Here we could just use string to replace `Enum`.
Once the required python version of janitor is set to 38, we just update documentation and typing